build.assets/Dockerfile-fips

```diff
@@ -78,5 +81,46 @@ RUN mkdir -p /opt && cd /opt && curl -L https://github.com/gravitational/libbpf/
 COPY pam/ /opt/pam_teleport/
 RUN make -C /opt/pam_teleport install
 
+ARG LLVM_VERSION=7.0.1
+RUN mkdir -p /opt && cd /opt && curl -L https://releases.llvm.org/7.0.1/clang+llvm-$LLVM_VERSION-x86_64-linux-gnu-ubuntu-18.04.tar.xz | tar xJ 
+ENV PATH="$PATH:/opt/clang+llvm-$LLVM_VERSION-x86_64-linux-gnu-ubuntu-18.04/bin"
+
+# Build FIPS certified BoringSSL library
+# https://csrc.nist.gov/projects/cryptographic-module-validation-program/certificate/3678
+ARG BORINGSSL_VERSION=ae223d6138807a13006342edfeef32e813246b39
+RUN mkdir -p /opt && cd /opt && \
+    git clone https://github.com/google/boringssl.git && \
+    cd boringssl && \
+    git checkout $BORINGSSL_VERSION && \
+    mkdir build && \
+    cd build && \
+    cmake -DCMAKE_C_COMPILER=clang -DFIPS=1 -DCMAKE_BUILD_TYPE=Release -GNinja .. && \
+    ninja
+
+# https://github.com/cloudflare/boring#support-for-pre-built-binaries
+ENV BORING_BSSL_PATH=/opt/boringssl
+
+# Install Rust
+ARG RUST_VERSION
+ENV RUSTUP_HOME=/usr/local/rustup \
+     CARGO_HOME=/usr/local/cargo \
+     PATH=/usr/local/cargo/bin:$PATH \
+     RUST_VERSION=$RUST_VERSION
+
+RUN mkdir -p $RUSTUP_HOME && chmod a+w $RUSTUP_HOME && \
+    mkdir -p $CARGO_HOME/registry && chmod -R a+w $CARGO_HOME
+
+# Install Rust using the ci user, as that is the user that
+# will run builds using the Rust toolchains we install here.
+# Cross-compilation targets are only installed on amd64, as
+# this image doesn't contain gcc-multilib.
```

Comment:
This comment doesn't apply here, and you should probably install `gcc-multilib` if you want to be able to cross-compile.